### PR TITLE
fetch: limit web streams usage in body mixin methods

### PR DESCRIPTION
**Pull request by mertcanaltin**
This PR aims to improve performance by optimizing the processing of webflows in the fetch module.
This experiment was already done by @KhafraDev , I just opened a pr

I thought there would be discussion about it and it might benefit fetch 🚀 
#2164
here is a benchmark made by @KhafraDev 🙏 
benchmarks:
```js
import { Response as UndiciResponse } from './index.js'
import { cronometro } from 'cronometro'

await cronometro({
  async 'undici Response' () {
    await new UndiciResponse('abc').text()
  },
  async 'global Response' () {
    await new Response('abc').text()
  }
})

```

```
╔═════════════════╤═════════╤═════════════════╤═══════════╗
║ Slower tests    │ Samples │          Result │ Tolerance ║
╟─────────────────┼─────────┼─────────────────┼───────────╢
║ global Response │   10000 │ 40169.19 op/sec │  ± 1.59 % ║
╟─────────────────┼─────────┼─────────────────┼───────────╢
║ Fastest test    │ Samples │          Result │ Tolerance ║
╟─────────────────┼─────────┼─────────────────┼───────────╢
║ undici Response │   10000 │ 44024.31 op/sec │  ± 2.12 % ║
╚═════════════════╧═════════╧═════════════════╧═══════════╝
```

### CODE REVIEW

**Comment by mertcanaltin** [comment]
I'm running tests, I'll fix them 🧪

File: lib/web/fetch/util.js [hunk 1]
```diff
@@ -1055,6 +1055,11 @@ async function fullyReadBody (body, processBody, processBodyError) {
   //    with taskDestination.
   const errorSteps = processBodyError
 
+  if (typeof body.source === 'string') {
+    successSteps(new TextEncoder().encode(body.source))
```
Comment (Uzlopak) [inline lib/web/fetch/util.js:1059]: I did not check your code and if this makes sense. But I think this wil break without-intl builds . check how we implemented utf8Decode and look if you can implement something for utf8Encode?!

Comment (mertcanaltin) [reply lib/web/fetch/util.js:1059]: thank you very much for your suggestion I will try this

Comment (tsctx) [inline lib/web/fetch/util.js:1059]: In this way, test failures can be corrected.
```suggestion
    const buffer = new TextEncoder().encode(body.source)
    successSteps(Buffer.from(buffer.buffer, buffer.byteOffset, buffer.byteLength))
```

Comment (mertcanaltin) [reply lib/web/fetch/util.js]: thank you for suggestion 🙏

Comment (mertcanaltin) [reply lib/web/fetch/util.js]: @tsctx I wonder how you caught this

Comment (tsctx) [reply lib/web/fetch/util.js]: Because I had the same error.

**Comment by KhafraDev** [comment]
There are related failures, but I don't necessarily think this should land as-is even if those failures were fixed.

**Review by KhafraDev — CHANGES_REQUESTED** [review]
(making sure to block this)

**Comment by mertcanaltin** [comment]
I wonder if you have a suggestion, if it's healthy to keep it as it is, I'll close this place @KhafraDev

**Pull request closed without merging**